### ISSUE TITLE
StandardScaler error when infering forecast on new data from TemporalFusionTransformer

### ISSUE THREAD

**Issue by SachaFabien** [jdb78/pytorch-forecasting#1273]
- PyTorch-Forecasting version: 0.10.3
- PyTorch version: 1.13.1
- Python version: 3.9.13
- Operating System: Windows 11

I am trying to infer forecast on new data and I keep running into this error when using TemporalFusionTransformer.predict method on a new data:
 
ValueError: Found array with 0 sample(s) (shape=(0, 1)) while a minimum of 1 is required by StandardScaler.

I tried passing a pandas dataframe with all the necessary features at first. Then I tried loading the training dataset and using the TimeseriesDataset.from_dataset method. Both run into the same issue.

Howerver if I create a new TimeseriesDataset from scratch I can run the predictions without issue.

Any idea as to what could cause this error?

**Comment by sairamtvv** [comment]
Did your dataframe include those future known covariates also . I mean the future values you are supposed to provide

**Comment by SachaFabien** [comment]
Yes it does, I use the same dataframe with both methods. Creating a new timeseriesdataset from the dataframe works fine but the predict method raises the error.
